plugins/GCodeReader/FlavorParser.py

```diff
@@ -368,6 +368,8 @@ def processGCodeStream(self, stream):
                     layer_number = int(line[len(self._layer_keyword):])
                     self._createPolygon(self._current_layer_thickness, current_path, self._extruder_offsets.get(self._extruder_number, [0, 0]))
                     current_path.clear()
+                    # start the new layer at the end position of the last layer
+                    current_path.append([current_position.x, current_position.y, current_position.z, current_position.f, current_position.e[self._extruder_number], LayerPolygon.MoveCombingType])
```

Comment:
I would also add this line when the gcode has a "changing tool" code.

Reply:
Can you do that, please? I am not that familiar with Python or this code so I will probably get it wrong.

Reply:
Yes, I'll do it 😄